src/attributes/classes.js

```diff
@@ -14,7 +14,23 @@ function getClass( elem ) {
 
 jQuery.fn.extend( {
 	addClass: function( value ) {
-		var classes, elem, cur, curValue, clazz, j, finalValue,
+		var classes = [];
+		var proceed = false;
+		if ( typeof value === "string" && value ) {
+			classes = value.match( rnothtmlwhite ) || [];
+			proceed = true;
+		} else if ( value.constructor === Array ) {
```

Comment:
Checking `constructor` doesn't work cross-realm (e.g. for arrays from an iframe), `Array.isArray` does. That's why we avoid `instanceof` checks. Checking `constructor` is even worse as that would not even work for `Array` subclasses.